### DISCUSSION TITLE
Proxy does not work with my PHP backend.

### DISCUSSION THREAD

**Discussion by SrinivasPrabhu794** [webpack/webpack#16887] [Q&A]
Hi All, I was experimenting today with my webpack devserver config and was trying to get data from my **`php backend running on port 80`**

I make calls in to my php backend like this from my ` react frontend which runs on port 3000`:
`const resp =  await fetch('/api/getXYZ.php');`

My proxy setting in my webpack config looks like this:
```
proxy: {
            '/api': {
                target: 'http://localhost',
                secure: false,
                changeOrigin: true
            },
        },
```

When I open  ` http://localhost/api/getXYZ.php` on my browser it returns me the JSON. But when the react app makes the request I get  this error

` [webpack-dev-server] [HPM] Error occurred while proxying request localhost:3000/api/getXYZ.php to http://localhost/ [ECONNREFUSED]`

I check the network tab and I see the 504 error instead.

So to check if the proxy itself was working or not I changed my code to point to some API on the web which returns a JSON. I made the following changes/including target changes to point to a different server.

**React:**
`const resp =  await fetch('/api/v2/users?size=2&is_xml=true');`

**webpack.config.json:** (Changed target)
```
proxy: {
            '/api': {
                target: 'http://random-data-api.com',
                secure: false,
                changeOrigin: true
            },
        },
```

and this changes seems to work from both browser and react. 

What is special about 'random-data-api.com' compared to 'localhost' ? Spent a few hours on this and not getting anywhere. Please help!!!

**Comment by alexander-akait** [comment]
Looks like your PHP application (or your web server) check your headers and check host in your headers...
